### ISSUE TITLE
Device turn on alone

### ISSUE THREAD

**Issue by LeoeLeoeL** [dresden-elektronik/deconz-rest-plugin#5626]
## Describe the bug
 There are some devices (bulbs and plugs) that perform a power reset and then turn on or off according their Power on/off configuration

## Expected behavior
 Normal behaviour of devices without "uncontrolled" turning on/off

## Environment

 - Host system: Raspberry Pi
 - Running method: Raspbian
 - Firmware version:  0x26720700
 - deCONZ version: v2.13.4-beta
 - Device: ConBee II
 - Do you use an USB extension cable: yes
 - Is there any other USB or serial devices connected to the host system? If so: Which? APC UPS

## deCONZ Logs
 
[18.55.12.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7796615/18.55.12.txt)

[13.28.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7796616/13.28.txt)

## Additional context
 the time in the title of the logs is when the issue appears

Devices involved are:
Osram Undercabinet TW Z3
iCasa ICZB-IW11D
Lidl smartplugs
Lidl bulbs

I don't know if there is something that trigs but all the dvices involved reset almost simultaneously.

It's not the 230V line in my house because Lidl bulbs linked to HUE bridge don't suffer this issue

**Comment by LeoeLeoeL** [comment]
One thing more.
When bulbs turn on after the reset Deconz/Phoscon reports still off.

**Comment by Mimiix** [comment]
Hi, 
I've edited the title as it is a bit subjective. Just like to keep it on point.

What log levels are involved here?

Kind regards,

**Comment by Mimiix** [comment]
Also, can you please add: A list of the nodes + their Mac ID and provide me a screenshot of your deCONZ gui.

**Comment by LeoeLeoeL** [comment]
Not so subjective. Devices that activate on their own; Think if somebody has a device to close/open the door to enter at home.
Logs involved are INFO and INFO 2.
The issue seems to increase slowly. It began with 2 devices (EKET BARS and SALA CINEMA) then the 3 DISIMPEGNO-1, then the plugs (AMPLI, SUBWOOFER and MOGE CAM) and now (after my first post) the Sonoff ZBMini called VIRTUAL SCALA SOPRA.
Now I moved temporarly to Windows. I imported the backup and I'm monitoring if issue still happens.
But for that I had to move the Conbbe II to another location in the hous (1 floor below) and in that position the coordinator lost 6 devices so that I don't know if this test will be reliable

**Comment by Mimiix** [comment]
> Not so subjective. Devices that activate on their own; 

I have yet to see a bug like this.  Never theless, i'm happy to explore it as a bug. But once we can determine it's setup related , i have to close it as it doesn't comply with #5113 .


Can you please provide me logs with:
INFO, INFO L2, ERROR , ERROR L2, APS, APS L2 and ZCL?

In addition, can you provide me the things requested earlier:
list of all nodes + Macs
Screenshot of the deCONZ gui.

**Comment by manup** [comment]
I'm not sure for Lidl devices, but with OSRAM/Ledvance we've already seen that at times they reboot on their own for unknown reasons. In that case you would see "Device Announce" messages in the logs.

**Comment by Mimiix** [comment]
> I'm not sure for Lidl devices, but with OSRAM/Ledvance we've already seen that at times they reboot on their own for unknown reasons. In that case you would see "Device Announce" messages in the logs.

I did see some of these. But i want to be sure. If that's the case ,and the other "affected" devices are routed trough these, i am closing the issue and forward the user to the forums.

**Comment by LeoeLeoeL** [comment]
This is the screen:
![image](https://user-images.githubusercontent.com/65825223/147947128-e0a5771d-7508-4464-a628-16844fbb3d7e.png)

Catching the log is not so easy because the "debug view" covers 7 minutes only. I have to be "so lucky" to see the reset while is performing. I hope asap.

**Comment by Mimiix** [comment]
> This is the screen: ![image](https://user-images.githubusercontent.com/65825223/147947128-e0a5771d-7508-4464-a628-16844fbb3d7e.png)
> 
> Catching the log is not so easy because the "debug view" covers 7 minutes only. I have to be "so lucky" to see the reset while is performing. I hope asap.

We still need the list of nodes and the ones that you say are affected.

**Comment by LeoeLeoeL** [comment]
I added again the Osram lamp (Eket Bars)
![image](https://user-images.githubusercontent.com/65825223/147954853-acf418bc-a9d3-4ac4-b2a2-c1d7d8407b1b.png)

![image](https://user-images.githubusercontent.com/65825223/147955211-88588dbf-5945-4d04-92aa-af7daf00e201.png)
![image](https://user-images.githubusercontent.com/65825223/147955270-acfbe622-de89-4dfa-a82d-59dd7dd38966.png)

Nodes involved are:
Eket Bars
Luce Sala Cinema
Luce Disimpegno C2 and C3 (C1 is under Hue Bridge now)
Ampli
Subwoofer
Luce Moge C
Moge Cam
Maybe Luce Cucina 1 and 2 too but it's difficult to say because they are under a physical switch; When they are OFF they can't turn on alone and when they are on they stay on after a reset due to configuration. But sometimes I see a very very fast dimming and brigtning .

**Comment by Mimiix** [comment]
It's really hard to see the nodes and how they are connected. Can you please make it easier to see connections between each of them?

**Comment by LeoeLeoeL** [comment]
I think everything is connected to the coordinator.
For instance, Moge Cam and Luce Moge C are located at "floor +1" and do not have any direct connection to Eket Bars, Ampli etc. that are located at "floor -1".

![image](https://user-images.githubusercontent.com/65825223/147959785-acb57999-3477-4d10-b0bf-c7538505dedd.png)

**Comment by LeoeLeoeL** [comment]
[19.05.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7803353/19.05.txt)

Event happened between 19.04 and 19.05. I think Eket Bars turned on few seconds before Luce Disimpegno-1 C2 and C3.
Luce Living C and Luce Moge C turned on too.

**Comment by LeoeLeoeL** [comment]
Home Assistant reports 19.05.33 for all the devices above. Ampli and Subwoofer too.

**Comment by manup** [comment]
The only requests towards the "Eket Bars" are neighbor queries, since reporting is intact the nodes attributes aren't queried.
The log otherwise has no indication of Device Announce messages.

What is noticeable here is that the overall mesh signal strength seems to be quite weak. Might be worth to check that the Zigbee channel doesn't interfere with WiFi / Bluetooth (streaming) nearby, the WiFi can be checked with a phone WiFi scanner app.

**Comment by Mimiix** [comment]
@manup the earlier logs did have device announcements

**Comment by manup** [comment]
Ahh ok, then it seems similar to the issue we saw in the other setup.

**Comment by LeoeLeoeL** [comment]
Channel is 25, It should be outside wifi.
I don't know how is possible to have stronger signal. The only green connections are among devices that "touch" each other.

**Comment by manup** [comment]
To get a better picture I'd suggest setting the display LQI filter higher, e.g. to 100:  "Panels > Source Routing > Minimum LQI Display"  this this hide all weaker than 100 LQI links.

>Channel is 25, It should be outside wifi.

Depends, for example WiFi channel 13 still overlaps. But it's best to look in a WiFi sniffer app on your phone to get a better picture, WiFi might also come from a neighbor :)

![image](https://user-images.githubusercontent.com/383386/147992481-c93a67f7-46c1-4140-a65e-3811a70b1ac8.png)

**Comment by LeoeLeoeL** [comment]
I'l try to set channels 1 and 6. But neighbours......
My Hue Bridge has channel 20 and works flawlessy.
I have another log.
[21.54.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7810393/21.54.txt)
This time only 3 devices are involved because the others were off (with poweron/off OFF) or on ( with poweron/off ON)
EKET BARS turned on alone and after 5-10 seconds Luce Disimpegno C2 and C3 followed.

**Comment by LeoeLeoeL** [comment]
A log with many events and all the device previously written in this thread.
[Many events.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7810595/Many.events.txt)
Eket bars is always the first

**Comment by manup** [comment]
From the logs there are several lights restarting, with MAC prefixes belonging to Ledvance/Osram and Nordic (here Vimar).
I don't see any hint that this could be caused by the coordinator. I checked the ASDU payloads that for ZCL reports the proper ZCL Default Responses are returned, but this looks fine.

The logs show that after each reboot the devices change their NWK addresses. Perhaps there is a bug in the lights firmware that they think there is an NWK address conflict, there was actually a case of this a few years ago but that was a different vendor.

Running out of ideas here, I think it's a problem which needs to be fixed in the lights firmware.

**Comment by LeoeLeoeL** [comment]
Fixes to so many different brands?
We are speaking about IKEA, Lidl (Tuya), iCasa and Osram. I can't say about Hue because these lamps are connected to hue bridge.

As soon as I can, when I catch it, I'll send an event triggered by Ikea bulbs when physically switched on. Maybe it could clarify better.

P.S. Which is the Vimar device involved? Finestra Dispensa?

**Comment by LeoeLeoeL** [comment]
Is it the same?
https://forum.phoscon.de/t/lights-turn-on-randomly/1299

**Comment by mabe1983** [comment]
I can confirm the same behavior with Ikea GU10 bulbs. I have this issue since I've upgraded Deconz to the latest version. In previous versions I never noticed any issues like this.

**Comment by WhimsySpoon** [comment]
I had a light group activate at 2am on the latest version and firmware on conbee ii

**Comment by LeoeLeoeL** [comment]
Sometimes the issue is "triggered" by a physical switch-on of an IKEA lamp ( after the switch-on I can see the lamp to dim and bright fast one time and then other devices star to switch) sometimes IMHO Deconz activates the devices; If I restart Deconz the problem quits to exist for a while.

**Comment by LeoeLeoeL** [comment]
This bug continues everyday many times per day.
[Again!.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/7963734/Again.txt)
Is it so impossible to understand where is the problem?
Because Deconz has a problem.
Do I have to buy a second Hue bridge to solve the problem myself?
I hope no.
Thanks in advance.

**Comment by WhimsySpoon** [comment]
This is continuing to happen for me too; usually overnight, although I had one earlier this morning. Unfortunately I wasn't able to grab the logs in time. So far it's happened to a light group, plus an individual light.

**Comment by GitHAMG** [comment]
Hello,
I have the same issue. I am running deconz inside Home Assistant, don't know if that is important or has something to do with it. My impression is that with the December? update (update of deconz to 2.13.04 and firmware update of ConBeeII) of the deconz plugin the issues started. Since then I have the same issues that suddenly lights turn on without any user activity (often during the night). We also have zigbee remote switches which worked flawlessly for months, and now since the update the connected lights switch on/off only after some random delay or not at all. I also have the impression that the connection between controller and devices is worse now (at least the display in deconz is far worse, much less "green" connections, seems to be same oberservation as for other issue above).
My wife gets really unhappy that the switches do no longer work reliably and that some lights switch on without interaction, it is bad if there is light during evening/night in a room where the blinds are still open. If there is no fix, I fear I have to switch to another gateway to not destroy WAF fully.
Thanks a lot to the whole team and thanks in advance for looking at the issue!

**Comment by WhimsySpoon** [comment]
I too am running deconz as an HA addon.

**Comment by WhimsySpoon** [comment]
Just happened again. Different light this time, again an Innr one. Deconz wasn't initially showing the status change to on.

**Comment by manup** [comment]
It's really hard to tell what is exactly going on, but if the lights reboot on their own this is a firmware issue of the lights. In contrast to my knowledge Hue lights don't have this problem at all. The Ikea lights used to have a bug where they completely stopped responding after receiving a certain amount of *Parent Announce* messages, which can be send by any router in the mesh, the error happens earlier the more end-devices are in the network which may switch between routers, causing these messages.

I'm not sure if this is still the case with recent Ikea firmware, or if they have added a watchdog to detect this and reboot the light, which isn't too bad as long as the previous state is preserved. In that regard it might be worth to try this mode in the *On/Off Cluster*:

![image](https://user-images.githubusercontent.com/383386/151684486-20f6ecfa-fb79-4f77-ab7e-1bd17888cdb0.png)

(double click on the attribute to change the configuration)

The commands we're sending to the lights haven't changed in a long time, it's basically querying the descriptors, neighbor tables, control and attribute query commands, nothing which should bring a light down.

As shown in above screenshot https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5626#issuecomment-1004215670, the mesh has a overall weak signal range (more red' colors). This can have various reasons as WiFi interference or streaming audio/video or simply the lights are too far apart.

Note that in recent deCONZ versions the coloring of the node links has changed to better show weak spots, this is only a visual change.

Here is how the network should look normally under healthy RF conditions (more green):

![image](https://user-images.githubusercontent.com/383386/151684575-5e1254f9-fe51-40d2-995f-ee9fef2c0c9b.png)

The coordinator or deCONZ can't influence this as this purely depends on the RF interference and distance between nodes.

You may also try to enable source routing under "Panels > Source Routing" but I'm afraid the RF problems should be addressed first to let it work properly, otherwise the routes would need to use quite low minimum LQI values.

**Comment by LeoeLeoeL** [comment]
It's long time my Deconz suffers this bug.
Ikea bulbs with new firmware continue to reset.
Osram smartplugs (ZLL) and bulbs (HA) reset too.
Lidl bulbs and smartplug reset.
I can agree with you about firmware problems but, I have 8 (eight) Lidl bulbs connected to a HUE bridge and the problem never happened to them.
So, if devices connected to ConbeeII reset while devices connected to HUE don't, there is a bug in Deconz.
Yesterday evening resets happened many times in few seconds. So, I installed the last Deconz available on phoscon.de in a new SD and restored a backup. I didn't have any reset until today at 9 o' clock AM.

**Comment by mabe1983** [comment]
Update: Three days ago I updated the firmware of all Ikea lamps to the latest version and switched on the source routing mode - unfortunately the problem still exists.  The problem now occurs with almost all Ikea lights ...and it's only happening since I've upgraded Deconz to version  2.13.04. Before this upgrade everything worked well for more than one year.

**Comment by WhimsySpoon** [comment]
I only have one Ikea light in my setup and it's unaffected. The lights that keep turning on are Innr GU10s and Innr Flex Lights. I tried changing the attribute as @manup suggests, but it's readonly on these devices.

I've owned the affected Innr lights for +3 years and this has only started happening recently. 

Some things to note:

- The Ikea light that I own was only added recently
- My WiFi was set to Channel 11 and Zigbee to channel 20 which overlap. I have since change my WiFi channel to 1. The LQIs have improved as a result

**Comment by LeoeLeoeL** [comment]
"PREVIOUS" could be a mitigation for lamps if there is only a reset but if you have many resets in seconds the lamps "flash".
"PREVIOUS" could be an useless mitigation for smartplugs; For instance, I have a videocam connected and, when the smartplug resets,  it reboots,

Regarding interacions between Wifi and Zigbee:
Conbee: CH 25
HUE: CH 20
Wifi: CH 1
I dont'think there are overlaps.

**Comment by manup** [comment]
>So, if devices connected to ConbeeII reset while devices connected to HUE don't, there is a bug in Deconz.

Unfortunately it's not that simple. As mentioned above deCONZ doesn't send any commands which could reset/reboot a light. It's all just standard ZCL commands. If the light firmware is robust it must work regardless at what commands are thrown at it, even unsupported or bogus commands – we are only sending standard commands...

If the light decides to reboot it could be an internal watchdog which catches a bug like hanging firmware, or perhaps this is done since the light thinks network doesn't work properly.

For a realistic comparison with Hue bridge you'd need to connect *all* your lights and sensors of mixed brands, which are currently connected deCONZ to the Hue bridge. We have already seen that battery powered end-devices can bring down Ikea lights when they change parents a lot due the Parent Announce messages. This can only be fixed by Ikea.

In a another issue we have seen in the sniffer that Osram lights rebooted and changed their NWK address, which they shouldn't (can also be seen in the logs) this caused sensors to reconnect to different parent, and after enough Parent Announce messages from various lights, the Ikea lights went downhill.

I'm not 100% sure if Ikea, Osram and Innr all use the same MCU vendor (Silabs) but I feel that there are bugs in those Zigbee stacks which aren't addressed yet. Philips also uses Silabs but they tend to test the hell out there devices and fix bugs in vendor stacks.

To match the Hue bridge exactly we could:
* Disable Neighbor Table queries, but then we loose the network map and source routing
* Not use source routing, but this becomes more important for networks > 50 nodes
* Only poll the lights instead using reporting, slower but possible. Note the Ikea bridge uses reporting too

Non of this would be pretty, and I'm afraid it would not fix the problem. Since these reboots only happen in some networks I think the problem is escalated by a mix of certain devices, but that's just a theory.

Here is a similar issue with a different USB dongle on zigbee2mqtt, which overall sends the same commands as deCONZ to the lights: https://github.com/Koenkk/zigbee2mqtt/issues/4878

To mimic the Hue bridge you could try:     (perhaps over night as a test)
* Disable neighbor table queries: click on the "CRE" button and uncheck "Routers and Coordinator"
* Disable all ZCL requests to a light: "Plugins > REST API Plugin > Uncheck Plugin Active"

(note at this point no control commands or queries would be done, only commands received from devices)

If the lights still reboots it would indicate it's not related to our commands, and vica versa.

**Comment by LeoeLeoeL** [comment]
I opened some time ago another issue when Sonoff ZBMini 1.0 ?) went in "pairing mode" alone. It could be the same problem.
To use them I had to migrate them to HUE Bridge. The strange thing is ZBmini 2.0 don't suffer that issue. Maybe could be useful to know the differences between the 2 releases.
Unfortunatly, it's not possible to update ZBminis with older firmware; Sonoff bridge doesn't have that option.

It's impossible to "replicate" all the mesh over HUE because:
1) HUE manages until 50 devices
2) HUE doesn't manage NON HUE sensors and switches.

You say Deconz doesn't reset our devices but we have evidence of that.
I understand the purpose of Deconz is to work with all the zigbee devices available in the world (while HUE, AQARA , SONOFF and TUYA can afford to take care their products only) but not at cost of reliability.

**Comment by manup** [comment]
>It's impossible to "replicate" all the mesh over HUE because:
>
>    HUE manages until 50 devices
>    HUE doesn't manage NON HUE sensors and switches.

Indeed, the limit can't be worked around but it is possible to join ZLL/ZB3.0 Zigbee devices to the Hue bridge during light search, sensors and switches won't show up in the app but they are in the network.

>You say Deconz doesn't reset our devices but we have evidence of that.

No, what I tried to say is that we *only* send standard commands, none of this should bring a Zigbee device down, a reboot command doesn't exist.

Oversimplified example, if we send an On command to a light, and it reboots/crashes, one could say it's a bug in deCONZ, I'd argue that the light firmware needs to be fixed.

This may sound silly: but an actual Ikea example from the past —  when a control command was send, while the light was dimming, the light crashed. For Ikea there are already a few workarounds implemented to mitigate light firmware bugs, but this isn't always possible. For example recently Ikea broke the groupcast feature in their remotes with an firmware update and there is nothing we can do about it :/

**Comment by LeoeLeoeL** [comment]
> Indeed, the limit can't be worked around but it is possible to join ZLL/ZB3.0 Zigbee devices to the Hue bridge during light search, sensors and switches won't show up in the app but they are in the network.

I don't know if it is really possible. I read about **touchlink** for that (but works for few devices). Anyway, if not present in the app, It would be not easy to unpair the devices in the Hue bridge later.

> To mimic the Hue bridge you could try: (perhaps over night as a test)
> Disable neighbor table queries: click on the "CRE" button and uncheck "Routers and Coordinator"
> Disable all ZCL requests to a light: "Plugins > REST API Plugin > Uncheck Plugin Active"
> (note at this point no control commands or queries would be done, only commands received from devices)

Does it mean I would lose the % of a window covering device?

**Comment by GitHAMG** [comment]
Hello all,
I have multiple devices being affected by sudden switch on on their own: Osram Tibea, outdoor lantern, outdoor Flex RGBW LED, E14 bulb, E27 bulb, and also a Paulmann 93999 (Zigbee Controller). They all worked fine - at least felt - until December update of deconz in Home Assistant.
I did a test now, I did a screenshot of my zigbee network in deconz inside Home Assistant (see picture 1). And then I used the same ConbeeII device and plugged it into a Windows PC (50 cm next to my Home Assistant PC), imported a backup of my Home Assistant deconz config, and then also did a screenshot (see picture 2). It looks way different I would say. The deconz version of Windows PC is 2.12.3.
![deconz01](https://user-images.githubusercontent.com/88974987/151711618-c7df8b14-df03-4a54-a026-96f581046697.JPG)
![windows02](https://user-images.githubusercontent.com/88974987/151711631-669393f0-5b55-4cf8-bfc6-9707e5e22e13.JPG)

**Comment by manup** [comment]
> I don't know if it is really possible. I read about touchlink for that (but works for few devices). Anyway, if not present in the app, It would be not easy to unpair the devices in the Hue bridge later.

No Touchlink needed here, for ZLL/ZB 3.0 devices the normal joining should work, when starting light search in the Hue app.

> Does it mean I would lose the % of a window covering device?

Devices won't get lost they are just not controllable during the test (same as when deCONZ is closed).

**Comment by LeoeLeoeL** [comment]
@ GitHAMG: From 2.13 the visualization of nodes changed.
We lost many green lines. :-(
I asked to have the possibility to choose between both visualization but...

> Devices won't get lost they are just not controllable during the test (same as when deCONZ is closed).
I'm speaking about the possibility to read the % open/close in the cluster.

**Comment by manup** [comment]
> Hello all, I have multiple devices being affected by sudden switch on on their own: Osram Tibea, outdoor lantern, outdoor Flex RGBW LED, E14 bulb, E27 bulb, and also a Paulmann 93999 (Zigbee Controller). They all worked fine - at least felt - until December update of deconz in Home Assistant. I did a test now, I did a screenshot of my zigbee network in deconz inside Home Assistant (see picture 1). And then I used the same ConbeeII device and plugged it into a Windows PC (50 cm next to my Home Assistant PC), imported a backup of my Home Assistant deconz config, and then also did a screenshot (see picture 2). It looks way different I would say. The deconz version of Windows PC is 2.12.3. ![deconz01](https://user-images.githubusercontent.com/88974987/151711618-c7df8b14-df03-4a54-a026-96f581046697.JPG) ![windows02](https://user-images.githubusercontent.com/88974987/151711631-669393f0-5b55-4cf8-bfc6-9707e5e22e13.JPG)

That's the visual change mentioned above, in recent versions deCONZ shows more realistic colors for links based on the minmum LQI value. A link always has two LQI values 1) how A sees B 2) how B sees A. You can click on the LQI button to inspect the numeric values, since these were more green in earlier versions I'd suspect there is a higher value and a very weak LQI value, like 60/200.

**Important:** this is a pure visual change to better show weak spots in the mesh, functionality isn't affected.

Good values are around 200 and above, anything lower than 170 should be considered problematic imho. Resons for low values are usually: interference as WiFi/Bluetooth, or simply too large distance between devices, or metal housings of lights.... everything that hinders the signal to propagate. The distance issue can be mitigated by putting routers between weak spots.

**Comment by Mimiix** [comment]
> I asked to have the possibility to choose between both visualization but...

That makes no sense and your just fooling yourself.. Just the threshold is changed. Nothing else

**Comment by manup** [comment]
> @ GitHAMG: From 2.13 the visualization of nodes changed. We lost many green lines. :-( I asked to have the possibility to choose between both visualization but...
> 
> > Devices won't get lost they are just not controllable during the test (same as when deCONZ is closed).
> > I'm speaking about the possibility to read the % open/close in the cluster.

While the plugin is disabled the values won't be updated in the REST-API but incoming attribute reports are shown in deCONZ Cluster Info panel.

**Comment by LeoeLeoeL** [comment]
> > I asked to have the possibility to choose between both visualization but...
> 
> That makes no sense and your just fooling yourself.. Just the threshold is changed. Nothing else

Sometimes there are nodes without lines. With the "alternate" view I could see if connections exist without entering in Source Routing and set 1

**Comment by Mimiix** [comment]
@manup just to get this thing sorted as this is getting derailed: is there a bug here?

**Comment by LeoeLeoeL** [comment]
You are right.
Let's refocus again on the real problem.

**Comment by Mimiix** [comment]
@LeoeLeoeL I suggest doing the test as @manup mentioned before.

**Comment by LeoeLeoeL** [comment]
I'll try.
Anyway it's very strange; From yesterday night (when I made a fresh installation) until this evening I had only one event at 09.46 AM.
Now, there are continuous events.

**Comment by WhimsySpoon** [comment]
An Innr GU10 downlight turned itself on in the bedroom at 2am this morning. Home Assistant showed the light as still being off, so it wasn't getting reported back to the API. I also noticed that when this happens, the lights are always in a "standard" white colour, suggesting they have rebooted in some way (they never come on at their previous state). 

I have a mixture of Hue and Innr lights, most of which I've owned for the past three years. I used to update the firmware on the Hue bulbs when I was using a Hue bridge 2+ years ago, but as far as I can remember, Innr have never made firmware updates available.

This issue has only started in the last month or so, at about the same time that I:
- Updated the Conbee II stick to the latest firmware
- Update Deconz to the latest version
- Introduced a single Ikea GU10 bulb into the mix (I've had Ikea smart plugs for 2/3 years without issue).

I'm going to remove the Ikea bulb today and see if it makes a difference.

**Comment by Mimiix** [comment]
@WhimsySpoon @LeoeLeoeL Can either of you please test what @manup Suggested?

**Comment by WhimsySpoon** [comment]
> @WhimsySpoon @LeoeLeoeL Can either of you please test what @manup Suggested?

Unfortunately the Innr bulbs do not support this as a writeable property - they are read only so I cannot enable what is described here (https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5626#issuecomment-1025055532)

I have enabled Panels > Source Routing, though.

**Comment by LeoeLeoeL** [comment]
I began yesterday night.
2 resets after the new configuration.

**Comment by Mimiix** [comment]
> I began yesterday night. 2 resets after the new configuration.

So even with the Plugin being disabled, the behavior occurs?

**Comment by LeoeLeoeL** [comment]
Affirmative.
Another event this morning at 9 AM.
Nobody was at home to trig the event switching something.

**Comment by LeoeLeoeL** [comment]
is it enough to deflag the square to disable rest api plugin or a reboot is mandatory?

**Comment by Mimiix** [comment]
Meaning the plugin is disabled at that point.

I've told Manuel. But if the plugin is disabled and you still see the behavior : it's the device firmware of the light.

**Comment by WhimsySpoon** [comment]
I've removed the single Ikea bulb from my network in the hope that it might make a difference.

What I fail to understand is how this has just suddenly become a firmware issue in random bulbs that have been rock solid for three years. Something else has to be involved.

**Comment by LeoeLeoeL** [comment]
The firmware of Osram, Tuya, and Ikea devices?
You are speaking about more than half of the devices in the houses of the people.
But, why is the the firmware good for Hue bridge? Daes it read a different firmware? :-)
If firmware were buggy I should see lamps to light on in my living room alone. 8 Lidl bulbs that work without problems.
I think developers must investigate about this bug deeper.

**Comment by Mimiix** [comment]
@LeoeLeoeL I have the same brands and don't see this behavior. Faulty firmware can do a lot to devices their packets are routed to. We know osrams can create malformed packages for example. 

Either way  I told you already this issue doesn't comply with #5113 so I'd like to stay on topic and constructive. I am all in for solving bugs. 

I'll wait for Manuel to reply again.

**Comment by LeoeLeoeL** [comment]
What probably I forgot to list: iCasa dimmer ICZB-IW11D has this behaviour too. It's the first to reset and then Lidl devices few seconds later. To test if it was the guilty I deleted from the mesh but the issue happened anyway.
Probably Lidl lamps light on later because their reboot is slower.
When the issue happens all devices reset at same time but sometimes some of them don't do. I don't know if because they are momentary disconnected from the mesh. For instance, Osram smartplug resets rarely.

OT to developers: when plugin is deactivated Aqara motion sensor doesn't work while multi sensor and contact sensor do.

**Comment by manup** [comment]
Very interesting result, we haven't had this info before. This hints in direction that devices decide on their own to make a reboot for some reason. If feasible could you please also test to let the network run with deCONZ closed, perhaps over night.

**Comment by WhimsySpoon** [comment]
With the ikea bulb removed from my network, I have not had any lights turning themselves on last night.

**Comment by GitHAMG** [comment]
I did a bit further testing. I noticed that the light with the best signal strength did not have attributes like vendor, model, etc. shown in deconz. Two other lights with same hardware had all attributes shown correctly. So I removed the device from the network, rebooted it, and added it back to my deconz zigbee network. 
For me it seems to be much better now. Reaction when a button on a switch is pressed is much faster and I also did not have light switch ons on their own since then without user action. 
As it was a short time since doing the change, I will continue to observe if swith ons occur again.
Thanks to te whole team for all the development and suppport!

**Comment by LeoeLeoeL** [comment]
> If feasible could you please also test to let the network run with deCONZ closed, perhaps over night

I have to move something else to HUE in order to test that otherwise my wife kick me out.

**Comment by LeoeLeoeL** [comment]
I rebooted Phoscon because I had continuous resets in seconds.
Ater that, half a hour passed, non resets until now.

**Comment by WhimsySpoon** [comment]
Another night with zero issues following the removal of the ikea bulb. Is it possible this device was sending bad packets over the network that was causing my existing devices to reboot?

**Comment by LeoeLeoeL** [comment]
What i did yesterday night:
- I deleted the last IKEA bulb from the mesh but the issue has continued.
- At 21.45 I closed Deconz and then switched off the RPi3 (where Deconz is installed).
- Multiple resets happened for the the following 5 minutes.
- Then devices involved in resets decreased.
- After 10 minutes from T0 all "resets" disappeared !
- Until 01.48 NO resets.
- At 01.48 I restarted Deconz.
- 2 minutes later devices began to reset again.

**Comment by manup** [comment]
Interesting, after switching off the RPi it seems that devices still tried to reach out to the gateway during the 5 minutes, perhaps since  they couldn't get through the reset happens (which they shouldn't do).

Side note: as soon as ConBee gets powered (or any gateway / Hue bridge / Ikea gateway) the gateway firmware broadcasts a message in the network like once per minute so that devices know the coordinator is there and to figure out a route towards it.

Question, when deCONZ is running are there other control commands send by a REST client? Like controlling lights via automation's or circadian light control?

For me it's unclear if the devices go bogus because they can't reach the gateway due the weak routing / signal quality or if they don't like some commands.

----------------------

Yesterday I connected a Osram GU10 for a few tests to my network and found out something interesting which I think is true for all Osram lights I've seen so far:

![image](https://user-images.githubusercontent.com/383386/152139425-078727c7-5736-4adf-8173-b8470cc4160d.png)

* The Osram GU10 and Ikea GU10 Spots 1, 2, 4 are only **40 cm** apart (LQI values should be around ~200 and up).
* The LQI values the Osram GU10 claims to see on the receiving side are always ca. 50 % lower than the sending LQI the other lights see.
 
I don't think the Osram lights are that bad receivers, compared to sending, especially at this distance. This looks more like a wrong calculation of LQI in the light firmware. The problem is that all surrounding lights take this low LQI value as base for routing to/over Osram lights.

**Comment by SwoopX** [comment]
Reading silently along with interest.

@manup I'd say you'd need at leasr a short sniffer	 log to understand what's actually going on. Would it maybe make sense to shut down the RPi again when it happens and then immediately flash the sniffer firmware and take a sniff?

**Comment by LeoeLeoeL** [comment]
Today, after deleting 2 more devices, I rebooted device in Deconz but resets continued.
Then I rebooted in Phoscon page.
After that (14 PM) no resets until 21 PM. Now, several resets in progress.

**Comment by mabe1983** [comment]
I've tried a few things over the past few days.  Frequencies checked and deconz set to another channel (25).  Connect the stick to the Raspberry with a 1m long USB extension cable (to avoid interference).  Downgrade to v2.12.6.  Source routing switched on and off.  All Ikea bulbs flashed with the latest firmware.  Checked signal quality (most connections are very good...few are average).  Unfortunately, I was not able to solve the problems with the mentioning listed above.  Some lights mostly turns on in the middle of the night (Ikea GU10, Ikea Driver, Ikea Float Pannel).  I have no problems with my Tint or Phillips Hue bulbs.  My network consists of about 60 devices (Ikea lightbulbs, lamps and drivers, Ikea blinds + switches and repeaters, Tint lightbulbs, Phillips lightbulbs, Aquara temperature and water sensors).  In the next few days I will try to remove the Conbee 2 and set up / pair the network with another ZigBee coordinator (CC2652P) - I will then report if the problems still exist.

**Comment by LeoeLeoeL** [comment]
Yesterday at 22 PM I deleted 2 more devices always involved (2 Lidl smartplugs).
Now I'm experiencing 24 hours without resets.
I don't think they were guilty. Probably it was a "contribution".

**Comment by LeoeLeoeL** [comment]
What does it mean when I see NWK Address changing as a slot machine?
Just few minutes ago I had several resets and one Vimar switch began to change that number.
Then there has been a "great" reset (the dimmer flashed 5 times ) and now it's quiet again with that Vimar that is maintaining the address.

**Comment by mabe1983** [comment]
Hey guys, I just wanted to give some feedback on my transition to a new coordinator.  I switched my entire Zigbee network to the new coordinator CC2652P (Firmware: 3.x.0 Z-Stack coordinator) last weekend.  Since the switch, I've had a damn stable network, no outages and, most importantly, the lights don't turn on by themselves.  In addition, I can send as many commands as I want at once and all lamps react immediately (that was not possible with the Conbee 2).  So I would strongly bet that something is wrong with the firmware of the Conbee Stick.

**Comment by Mimiix** [comment]
@mabe1983 is the network exactly the same? All same devices?

**Comment by LeoeLeoeL** [comment]
The way to determine if Conbee firmware is ok or not could be to use it with ZHA. 
By the way, it's long time Deconz has this behaviour but the resets happened few times; Now the network becomes crazy.

**Comment by Mimiix** [comment]
@mabe1983 let's see what manup says about this. Really amazed here :)

**Comment by manup** [comment]
That's indeed super interesting, so this brings the question what's the difference? In theory absolutely nothing should bring a light down as there is no reboot command, but something must be different here.

@mabe1983 would you mind sharing a sniffer log of the CC network for are more low level comparison and provide it to me mpi@dresden-elektronik.de?

Sniffing can be done with ZShark and the ConBee:

https://phoscon.de/en/conbee2/software#zshark

**Comment by LeoeLeoeL** [comment]
Compression and moiree "destroyed this video but you can see how many times that device changes its address in 12 seconds.
https://user-images.githubusercontent.com/65825223/153259100-ce24387f-fe5e-431c-b5c5-1313527ee1f8.mp4
Is it the cause or the effect of the continuos resets of devices?

**Comment by Mimiix** [comment]
@LeoeLeoeL well, it does cause spam in the network. Routers parsing these might be some sort of ddossed by this.

What happens if you remove it from the net?

**Comment by LeoeLeoeL** [comment]
> What happens if you remove it from the net?

Another device begins to make the same

**Comment by manup** [comment]
That's pretty hardcore, usually the NWK address changes on routers happen if they think there is an address conflict and they get a new NWK address to resolve it (Zigbee stochastic NWK addressing).

However, I dived a bit deeper in the Zigbee stack library and found a suspicious hacky piece of code which could potentially be the root here and trigger false address conflicts within the broadcast table handling. I have it disabled the lines in the following firmware, can you please give it a try:

(zipped as GitHub doesn't allow uploading custom file types)

[deCONZ_ConBeeII_0x26740700.bin.GCF.zip](https://github.com/dresden-elektronik/deconz-rest-plugin/files/8037502/deCONZ_ConBeeII_0x26740700.bin.GCF.zip)

**Comment by WhimsySpoon** [comment]
Amazing investigative work!

Is it worth me reintroducing my IKEA bulb into my network to see if the firmware bug is responsible for that too?

**Comment by LeoeLeoeL** [comment]
Just updated.
Now I can only see visual connections "very slow forming". After 10 minutes only 3 devices.
But, even if there isn't the visual link, device is reachable by Home Assistant.
It's curious the first device the Coordinator connected is the device I showed in the video yesterday.

**Comment by LeoeLeoeL** [comment]
Changes are easy to see when Deconz (for some reason) doesn't give the name to the device.
I tried to use NODE LIST but it's a static table; If the address changes after the list is opened  the value is not updated.
After a Deconz reinstallation (and a firmware update could be assimilates as an installation) I experienced until 24 hours of "peace" (no resets).
So that, only after 48 hours without reset I can only scream URRA'!!!!

**Comment by Mimiix** [comment]
Let's hope it stays that way! @LeoeLeoeL

**Comment by LeoeLeoeL** [comment]
first reset 15 minutes ago

**Comment by LeoeLeoeL** [comment]
Yes, they changed their addresses as in my videoclip.

**Comment by LeoeLeoeL** [comment]
Yesterday evening I deleted from the network 2 Vimar switches (still there are 5 Vimar covering devices). One of them was the guilty of changing address quickly.
After that, no more resets.
Now question is:
Is really that device responsible for the issue? Other people who experienced the same don't have Vimar but Ikea or something else.
The issue began before I bought those device even if it wasn't so "strong".

**Comment by LeoeLeoeL** [comment]
In absence of resets, from tomorrow evening I add one device each 24 hours to see if/when resets will happen again.

**Comment by LeoeLeoeL** [comment]
I'm trying to add a bulb in the network but I can't !
Is it a bug of the beta firmware?

**Comment by LeoeLeoeL** [comment]
While waiting for a confirmationa regarding pairing with the new beta firmware,
after 36 hours of "peace", a Vimar covering device started to change address simultaneously with the resets.

**Comment by LeoeLeoeL** [comment]
After the short "reset-storm" of yesterday afternoon (my previous post) when the Vimar covering device looked for an address to its liking, no more resets.
Is the network settled ?
Few devices few problems?
Until now I discovered Vimar devices only that change address but I'm not sure if others do that.
When FW is fixed for joining  I'll add again 2 Vimar switch to see if there will be different behaviour in resets between the beta FW and the last official one.

**Comment by manup** [comment]
Damn, the stack tracks if a neighbor is in address conflict in a variable for each neighbor. But it wasn't initialized properly so could be 0 or 1. I've fixed this now and the join issue in 0x26750700. So far this looks good in the sniffer no address conflicts,  I'll keep it running over night to see if anything pops up.

[deCONZ_ConBeeII_0x26750700.bin.zip](https://github.com/dresden-elektronik/deconz-rest-plugin/files/8062764/deCONZ_ConBeeII_0x26750700.bin.zip)

**Comment by manup** [comment]
Interestingly in both cases the devices (here Xiaomi) did send a malformed ZCL Attribute Report 

![image](https://user-images.githubusercontent.com/383386/153921786-8ddd6fa9-be65-4b23-bfad-87efb23a2f03.png)

**Comment by LeoeLeoeL** [comment]
I just updated.
A question could be , why don't your device reset?

**Comment by manup** [comment]
I think they don't get the message as they already sleep when it is send. It seems that this happens when the MAC address is not yet set in the neighbor table, during the first MAC Poll Data Request. As soon as the Report is received we get the false positive:

NWK 1234  MAC 0
NWK 1234 MAC 1111...   (conflict: 0 is not 1111)

Here the MAC address of 0 should just be replaced, next I'm going to check if this is the issue.

**Comment by LeoeLeoeL** [comment]
Trying to pair again the Vimar switch.

https://user-images.githubusercontent.com/65825223/153966155-65a8ffd8-5d34-4295-ae4a-761a94c28a8c.mp4

5 minutes looking for an address. After 5 minutes pairing time expires and Deconz keeps the label with one circle only.

**Comment by manup** [comment]
Interesting is this with version 0x26760700?

I kept the sniffer running over night here, didn't see any address conflicts anymore.
One difference in my case is, they happened for end-devices.

**Comment by LeoeLeoeL** [comment]
Yes, 0x26760700.
Also the other Vimar switch label was present with one circle even if I deleted it many times and not paired again yet.

**Comment by LeoeLeoeL** [comment]
With  this FW is very difficult to add devices. 
Anyway, resets still happen; now it's another Vimar device that is looking for a new address. 2 options after this behaviour:
1) Device gets the address and there is a peace in the network for a while.
2) Device is kicked out temporarily (the label in VNC becomes grey.

**Comment by LeoeLeoeL** [comment]
This firmware is a disaster for the device added after flashing it.
Device are paired and then lost. Sometimes greyed and sometimes they lose the name on the label.

**Comment by manup** [comment]
While device is rebooting constantly deCONZ has no chance to query anything from it. I don't know what's so different on Vimar devices for this to be triggered, looking forward to order one to check in the sniffer if the problem can be reproduced.

The following firmware doesn't contain any address conflict resolution, if the device still reboots with this the cause would be something different entirely.

[deCONZ_ConBeeII_0x26770700.bin.zip](https://github.com/dresden-elektronik/deconz-rest-plugin/files/8088323/deCONZ_ConBeeII_0x26770700.bin.zip)

**Comment by LeoeLeoeL** [comment]
I forgot to mention Deconz reboots with 26760700.
Update: Also other device brands changed address but I don't know if due resets od drvices or deconz reboots.
Yesterday evening:
![image](https://user-images.githubusercontent.com/65825223/154521559-300326cd-7308-4279-8e5f-8f8cc7002ad7.png)

Now:
![image](https://user-images.githubusercontent.com/65825223/154521701-a566c3a8-d712-4e2e-ae2c-b4457e507644.png)

**Comment by LeoeLeoeL** [comment]
Yesterday afternoon after 3 days without events (with FW 26770700) I began to add some devices.
More than some :-)
3 Lidl plugs
4 Lidl bulbs
1 Sonoff ZB Mini 1.0
1 Ledvance Undercabinet TW Z3
1 Ikea bulb
In the evening I had an event so I deleted the last device paired (Ikea bulb). In the night (2 a.m.) I had another event.
So, with the new firmware there aren't "storms of resets" but the problem still exists.
During the 3 days, some devices changed address (Sonoff, Vimar and Ikea repeaters) but without resetting. There weren't Lidl devices in the network.
Events happen when Vimar devices look for an address as in the video shown above.

**Comment by manup** [comment]
That's quite a device collection :) Interesting that some devices still change their NWK addresses, at least we know that it isn't caused by address conflicts from the gateway. But still a big mystery what is going on. I'm trying to get a mains powered Vimar device to hopefully spot something in the sniffer.

>A very stupid question: As for IP addresses, is it possible to make static NWK addresses?

In theory yes, but almost all devices manage their own NWK address which they can change if they think an address conflict exists (stochastic addressing).

**Comment by LeoeLeoeL** [comment]
I'm becoming crazy.
This afternoon I had 3 or 4 events.
One Vimar started to look for a new address.
But the strangest thing is now I have in my network 2 devices I haven't!!!
![image](https://user-images.githubusercontent.com/65825223/155023952-87535453-9fba-4c19-b62f-448196bbe020.png)
![image](https://user-images.githubusercontent.com/65825223/155024018-53ee18f7-6507-4a76-8c28-cd126d7ca326.png)
I'don't have Ikea transformers and I didn't enable pairing mode but:
![image](https://user-images.githubusercontent.com/65825223/155024213-47941741-72cd-4281-b8bb-38dbbb845145.png)
Deconz goes in pairing mode alone!

**Comment by manup** [comment]
>Deconz goes in pairing mode alone!

:thinking:  it can't, this can only be done by manually pressing the button or externally via REST-API.

Can you please enable the LQI button to see how the network quality is around the Vimar devices?

**Comment by manup** [comment]
Just to be sure this is with version v2.14.1?

**Comment by LeoeLeoeL** [comment]
But it did. I watched the countdown without any click.
![image](https://user-images.githubusercontent.com/65825223/155025942-477a3924-4854-4718-be56-9fb158491c8e.png)
Another issue is (llast FW or 2 last firwares) Deconz disconnects and reconnects itself. Often when I move devices in VNC.

![image](https://user-images.githubusercontent.com/65825223/155026855-2ed26016-a952-43b7-b935-9f586edca3fa.png)

**Comment by LeoeLeoeL** [comment]
24 hours without events.
Ikea devices are present in the network even if they are not mine! Does my neighbour know about those???
Regarding the problem of pairing mode "alone", perhaps the issue is in the Phoscon page.

**Comment by LeoeLeoeL** [comment]
Almost 48 hours without resets.
Then, suddenly, 2 resets while a Vimar device looked for a new address. This time that device became grey after a minute and the quietude came back.
The positive thing is with the last firmware the resets have decreased.

**Comment by LeoeLeoeL** [comment]
A nice update.
From the last post, in the first 3 days I had 2 resets with few devices. After those, no more resets.
3 Ikea lights are in the network again and I added one more Osram smartplug and 2 3A Smart Home DE model LXN59-2S7LX1.0.
Now the network is bigger than before when it suffered lots of resets.
I activated source routing too.
![image](https://user-images.githubusercontent.com/65825223/156812015-ef303632-0225-42ab-8f36-b1d89fc5a560.png)

I have some to report (probably due to beta firmware)
1) When Deconz/Phoscon is in pairing mode, the pairing time is much longer than before.
2) In VNC, when I move a device, sometimes Deconz disconnects and reconnects.
3) When source routing is ON, Deconz understands later when a device is OFF due to a POWER OFF. Amost 15 minuts. I tested with Ikea bulbs; Normally they take longer time respect other devices in standard routing too but no more than 6/7 minutes.
4) When a device is OFF, due to a POWER OFF, the source route link still remains.

I cross the fingers :-)

**Comment by WhimsySpoon** [comment]
This is great news. Do you still have the devices that were suffering the resets in your network?

**Comment by LeoeLeoeL** [comment]
Yes, all the "crazy" devices are in the network.

**Comment by WhimsySpoon** [comment]
This IS great news! Might try the beta firmware with my troublesome Ikea light this weekend.

**Comment by keesj** [comment]
Hi coming from https://github.com/zigpy/zigpy/issues/854#issuecomment-1036532442 
I also have a few IKEA lights I sopped using marked "BAD" due to blinking issues. I am willing to spend some time testing out things. My setup is conbee II with home assistant the zha

**Comment by manup** [comment]
@keesj you might try also the above mentioned firmware in https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5626#issuecomment-1042864121 if it improves your setup.

**Comment by LeoeLeoeL** [comment]
18 days without events! (It seems something said in the Alcoholics Anonymous :-) )
Just updated to 2.15.0 beta.
What about the firmware 0x26770700 I have in the Conbee II? Will it be "upgraded" to official or a special one for people who have problems of events?

**Comment by LeoeLeoeL** [comment]
Very strange behaviour.
Yesterday afternoon I had to disconnect from current all the "light line" at home due to change a wall switch.
Then I powered on the "light line" again.
After that, events come back.

**Comment by LeoeLeoeL** [comment]
The issue described in the thread seems to be fixed by firmware 0x26770700.
As I wrote before, the network quits to have strange devices behaviour after 2 days since the installation or... a blackout.

@manup: What about the firmware 0x26770700 I have in the Conbee II? Will it be "upgraded" to official or a special one for people who have problems of events? Will I experience a regression in case of a new public release?

**Comment by manup** [comment]
Hi, yes the fixes in 0x26770700 will also be part of the upcoming official version, so it's safe to update when released.

**Comment by WhimsySpoon** [comment]
Is the new firmware being released soon? I didn't see it as a part of the recent release and I'm still having lights turn on in the middle of the night.

**Comment by Nama** [comment]
If my router restarts (network not available for some minutes) or I restart the deconz service, the ikea bulbs turn on as deconz starts.
deconz and node-red are on the same RPi with the Raspbee II. So no idea why the network effects it.
Is my issue related to this issue?

**Comment by LeoeLeoeL** [comment]
Sometimes the problem reappears but doesn't involve so many devices as before.

**Comment by LeoeLeoeL** [comment]
Updated to 2.17.0 beta last evening.
2 events occurred during the night

**Comment by LeoeLeoeL** [comment]
> Hi, yes the fixes in 0x26770700 will also be part of the upcoming official version, so it's safe to update when released.

I updated to 0x26780700.
I have a regression.
Many events again.

**Comment by LeoeLeoeL** [comment]
I think this is one of the hardest issues to fix.
From yesterday afternoon no events.

**Comment by WhimsySpoon** [comment]
I re-introduced the Ikea GU10 lamp that was causing my issues a few weeks ago, haven't had a single blip since. The only thing that's changed is I've been keeping deconz updated (via the Home Assistant add-on). I haven't applied any firmware updates to the Conbee II stick.

**Comment by Mimiix** [comment]
@haraldknoll could be anything. Please open a topic on the forums to check your case.
